packages/@glimmer/debug/lib/dism/opcode.ts

```diff
@@ -0,0 +1,306 @@
+              case 'SyntaxError':
+              case 'TypeError':
+              case 'WeakMap':
+              case 'WeakSet':
+                return `<${name}>`;
+              case 'Object':
+                return `<${name}>`;
+            }
+
+            if (value instanceof Map) {
+              return `<Map[${value.size}]>`;
+            } else if (value instanceof Set) {
+              return `<Set[${value.size}]>`;
+            } else {
+              return `<${name}>`;
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+export function debugValue(item: unknown, options?: ValueRefOptions): Fragment {
+  if (isIndexable(item)) {
+    const classified = getLocalDebugType(item)!;
+
+    if (classified) return describeValue(classified);
+  }
+
+  return unknownValue(item, options);
+}
+
+function describeValue(classified: ClassifiedLocalDebug): Fragment {
+  switch (classified.type) {
+    case 'args':
+      return describeArgs(classified.value);
+
+    case 'args:positional':
+      return positionalArgs(classified.value);
+
+    case 'args:named':
+      // return entries
+      return namedArgs(classified.value);
+
+    case 'args:blocks':
+      return frag`<blocks>`;
+
+    case 'cursor':
+      return describeCursor(classified.value);
+
+    case 'block:simple':
+    case 'block:remote':
+    case 'block:resettable':
+      return describeBlock(classified.value, classified.type);
+
+    case 'factory:helper':
+      return Fragment.special(classified.value);
+
+    case 'syntax:source':
+      return describeSyntaxSource(classified);
+
+    case 'syntax:symbol-table:program':
+      return describeProgramSymbolTable(classified);
+
+    case 'syntax:mir:node':
+      return describeMirNode(classified);
+  }
+}
+
+function describeArgs(args: VMArguments) {
+  const { positional, named, length } = args;
+
+  if (length === 0) {
+    return frag`${as.type('args')} { ${as.dim('empty')} }`;
+  } else {
+    const posFrag = positional.length === 0 ? empty() : positionalArgs(positional);
+    const namedFrag = named.length === 0 ? empty() : namedArgs(named);
+    const argsFrag = join([posFrag, namedFrag], ' ');
+
+    return frag`${as.type('args')} { ${argsFrag} }`;
+  }
+}
+
+function positionalArgs(args: PositionalArguments) {
+  return join(
+    args.capture().map((item) => value(item)),
+    ' '
+  );
+}
+
+function namedArgs(args: NamedArguments) {
+  return join(
+    Object.entries(args.capture()).map(([k, v]) => frag`${as.kw(k)}=${value(v)}`),
+    ' '
+  );
+}
+
+function describeCursor(cursor: Cursor) {
+  const { element, nextSibling } = cursor;
+
+  if (nextSibling) {
+    return frag`${as.type('cursor')} { ${as.kw('before')} ${Fragment.special(nextSibling)} }`;
+  } else {
+    return frag`${as.type('cursor')} { ${as.kw('append to')} ${Fragment.special(element)} }`;
+  }
+}
+
+function describeBlock(
+  block: AppendingBlock,
+  type: 'block:simple' | 'block:remote' | 'block:resettable'
```

Comment:
simple blocks were [renamed to appending blocks](https://github.com/glimmerjs/glimmer-vm/pull/1663/files#diff-d81a95325f353a0a60d137f71a5a57f6f4ef3da2a4dad75069c65744035ade47R99), yeah?

what determines simple vs remote vs resettable?

Reply:
- Appending: A block that will never be cleared. However, since it's possible to append a block to an Appending Block, the boundaries of an Appending Block can change. See `FirstNode` / `LastNode` for details on how this works.
- Resettable: A block that can be reset (i.e. cleared). When a block is reset, its contents are cleared and it produces a new cursor to render a block into. This is used for constructs like `{{#if}}` / `{{else}}`, which clear and rerender the block whenever the boolean value changes.
- Remote: A block that represents a remote element. When the block is destroyed, the nodes rendered into the remote element are removed.